core/dbt/contracts/graph/manifest.py

```diff
@@ -793,6 +793,8 @@ def deepcopy(self):
             files={k: _deepcopy(v) for k, v in self.files.items()},
             state_check=_deepcopy(self.state_check),
         )
+        copy.build_flat_graph()
```

Comment:
I think instead of building this again, the idea was that we'd do a copy of the flat_graph dictionary, like the nodes/sources/macros/files/state_check above. So add a line like:  flat_graph={k: _deepcopy(v) for k, v in self.flat_graph.items()},

Reply:
On second thought, this is probably better than doing a deepcopy of the flat graph, which will make yet another copy of the nodes and is probably slower.